### PR TITLE
Pdb dataset fix

### PR DESCRIPTION
**Pull request by kierandidi**
Fix the PDB dataset to 
- avoid redownloading the whole dataset if it is already present on disk
- remove import issues and unpacking problems in the download method

### CODE REVIEW

File: proteinworkshop/datasets/pdb_dataset.py [hunk 1]
```diff
@@ -157,19 +165,45 @@ def __init__(
...
+                v["id"] = v["pdb"] + "_" + v["chain"].str.join("")
+                log.info(v)
+                splits[k] = v.loc[v.id.isin(ids_to_exclude) == False]
+                log.info(
+                    f"Split {k} has {len(splits[k])} chains after excluding failing PDB"
+                )
+        self.splits = splits
+        breakpoint()
+        return splits
+    # def parse_dataset(self) -> pd.DataFrame:
```
Comment (amorehead) [inline proteinworkshop/datasets/pdb_dataset.py:189]: Perhaps these commented lines can be removed?

Comment (kierandidi) [reply proteinworkshop/datasets/pdb_dataset.py]: done, thanks for the heads up!

File: proteinworkshop/datasets/pdb_dataset.py [hunk 2]
```diff
@@ -130,21 +124,28 @@ class PDBDataModule(ProteinDataModule):
...
         self,
         path: Optional[str] = None,
+        structure_dir: Optional[str] = None,
         pdb_dataset: Optional[PDBData] = None,
         transforms: Optional[Iterable[Callable]] = None,
         in_memory: bool = False,
         batch_size: int = 32,
         num_workers: int = 0,
         pin_memory: bool = False,
+        structure_format: str = "mmtf.gz",
```
Comment (a-r-j) [inline proteinworkshop/datasets/pdb_dataset.py:134]: Can you update the relevant config file to ensure it has all args present.

File: proteinworkshop/datasets/pdb_dataset.py [hunk 3]
```diff
@@ -130,21 +124,28 @@ class PDBDataModule(ProteinDataModule):
...
         self.root = path
         self.dataset = pdb_dataset
         self.dataset.path = path
-        self.format = "mmtf.gz"
+        self.format = structure_format
         self.overwrite = overwrite
 
+        if structure_dir is not None:
+            self.structure_dir = pathlib.Path(structure_dir)
+        else:
```
Comment (a-r-j) [inline proteinworkshop/datasets/pdb_dataset.py:146]: We may need to create these directories if they do not already exist.

**Review by a-r-j — CHANGES_REQUESTED** [review]
LGTM, couple of nits. Appreciate the PR!

File: proteinworkshop/datasets/pdb_dataset.py (hunk 2 above)
Comment (kierandidi) [reply proteinworkshop/datasets/pdb_dataset.py:134]: done @a-r-j

File: proteinworkshop/datasets/pdb_dataset.py (hunk 3 above)
Comment (kierandidi) [reply proteinworkshop/datasets/pdb_dataset.py:146]: done @a-r-j

**Review by a-r-j — APPROVED** [review]

**Pull request merged**